### PR TITLE
machine: compute sha256 as we are reading the file

### PR DESCRIPTION
**Pull request by guillaumerose**
small refactoring that can help a bit.

### CODE REVIEW

File: pkg/machine/fcos.go [hunk 1]
```diff
@@ -91,24 +90,19 @@ func UpdateAvailable(d *Download) (bool, error) {
...
-			if err != nil {
-				return false, err
-			}
-			s := sha256.Sum256(b)
-			sum := digest.NewDigestFromBytes(digest.SHA256, s[:])
-			if sum.Encoded() == d.Sha256sum {
-				return true, nil
-			}
-		}
+	defer fd.Close()
```
Comment (baude) [inline pkg/machine/fcos.go:100]: can you catch the error and use logrus to dump it?

**Comment by baude** [comment]
lgtm, just the one suggestion

File: pkg/machine/fcos.go (hunk 1 above)
Comment (guillaumerose) [reply pkg/machine/fcos.go]: yes done

**Comment by mheon** [comment]
/lgtm

Nice work!

**Pull request merged**